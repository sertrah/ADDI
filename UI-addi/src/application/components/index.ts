export { default as AppLayout } from "./AppLayout";
export { default as Loader } from "./Loader";
export { default as InputText } from "./InputText";
export { default as Card } from "./Card";
export { default as Button } from "./Button/Button";
export { default as NotificationMessage } from "./NotificationMessage";
export { default as Container } from "./Container";
