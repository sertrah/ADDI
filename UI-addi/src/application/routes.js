import ListLead from "presentation-layer/lead/ListLead";
import AddLead from "presentation-layer/lead/AddLead";
import ListProspect from "presentation-layer/prospect/ListProspect";
import Home from "presentation-layer/home";

import { ROUTER_PATH_LIST } from "./constants";

const routes = [
  {
    path: ROUTER_PATH_LIST.default,
    component: Home,
  },
  {
    path: ROUTER_PATH_LIST.leads,
    component: ListLead,
  },
  {
    path: ROUTER_PATH_LIST.addLead,
    component: AddLead,
  },
  {
    path: ROUTER_PATH_LIST.prospects,
    component: ListProspect,
  },
];

export default routes;
