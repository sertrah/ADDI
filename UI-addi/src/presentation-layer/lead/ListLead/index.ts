import ListLead from "./ListLead"

export default ListLead;