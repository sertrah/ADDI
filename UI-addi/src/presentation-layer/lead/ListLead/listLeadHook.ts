import React, { useState } from "react";
import { useNotification } from "application/hooks";
import { commonService, leadService } from "infraestructure/services";
import { useMutation, useQueryClient   } from "react-query";

export const useListLeadBussinessLogic = () => {
  const notify = useNotification();
  const queryClient = useQueryClient()
  const [loading, setLoading] = useState("");
  const { mutate: updateLead } = useMutation(leadService.update, {
    onSuccess: () => {
      notify.success({
        title: "Lead",
        message: "In good time, this lead was updated",
      });
      queryClient.refetchQueries()
    },
    onError: (error) => {
      notify.error({
        title: "Lead",
        message: error,
      });
    },
  });


  const checkLeadScore = (id: string) => {
    setLoading("Obteniendo score");
    commonService.getScoreById(id).then(({ score }) => {
      setLoading("");
      if (score >= 60) {
        updateLead({id, type: "prospect"})
      } else {
        notify.error({
          title: "Lead",
          message: `"Ups... su puntaje fue de ${score}, no paso el minimo`,
        });
      }
    });
  };

  const validLeadRecords = (id: string) => {
    setLoading("Verificando datos");
    Promise.all([
      commonService.validJudicialRecordsByID(id),
      commonService.validNationalRegistryByID(id),
    ]).then((responses) => {
      if (responses.every((x) => x.isValid)) {
        checkLeadScore(id);
      } else {
        setLoading("");
        notify.error({
          title: "Lead",
          message: "Ups... este lead parece tener algo sospechoso.",
        });
      }
    });
  };
  return { validLeadRecords, loading };
};
