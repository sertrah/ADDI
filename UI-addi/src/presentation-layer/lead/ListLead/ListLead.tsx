import React, { FunctionComponent, useState } from "react"; // importing FunctionComponent
import { useQuery } from "react-query";
import { leadService } from "infraestructure/services";
import { Card, Loader } from "application/components";
import { useListLeadBussinessLogic } from "./listLeadHook";
import { IClient } from "application/types";

const ListLead: FunctionComponent = () => {
  const { data: leadList, isLoading } = useQuery(
    "products",
    leadService.getAll,
    {
      retry: 1,
      retryDelay: 3000,
    }
  );
  const { validLeadRecords, loading } = useListLeadBussinessLogic();

  return (
    <section>
      <h1>List of qualified lead</h1>
      {loading && <Loader text={loading} />}
      {!isLoading &&
        leadList?.map((lead: IClient) => (
          <Card
            record={lead}
            btnAction={() => validLeadRecords(lead.id)}
            btnText="Validar lead"
            key={lead.id}
          />
        ))}
    </section>
  );
};

export default ListLead;
