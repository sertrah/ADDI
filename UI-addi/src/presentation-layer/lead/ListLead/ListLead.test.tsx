import React from "react";
import { render, screen } from "@testing-library/react";
import ListLead from "./ListLead";
import { QueryClientProvider, QueryClient } from "react-query";
import { NotificationProvider } from "application/contexts";

const queryClient = new QueryClient();

test("renders Home title", () => {
  const component = render(
    <QueryClientProvider client={queryClient}>
      <NotificationProvider>
        <ListLead />
      </NotificationProvider>
    </QueryClientProvider>
  );
  expect(component.container).toMatchSnapshot();
});
