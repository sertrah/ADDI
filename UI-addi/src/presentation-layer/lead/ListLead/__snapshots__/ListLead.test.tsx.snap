// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home title 1`] = `
<div>
  <section>
    <h1>
      List of qualified lead
    </h1>
    
  </section>
</div>
`;
